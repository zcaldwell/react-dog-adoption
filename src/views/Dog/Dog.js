import React from 'react';
import { useState, useEffect } from 'react';
import { fetchDogById, deleteDog } from '../../services/dogs';
import DogDetail from '../../components/DogDetail';
import { useParams, useHistory } from 'react-router-dom';

export default function Dog() {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const params = useParams();
  const history = useHistory();

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(params.id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  const handleEdit = async () => {
    history.push(`/dogs/${params.id}/edit`);
  };

  const handleDelete = async (e) => {
    e.preventDefault();
    try {
      await deleteDog(params.id);
      alert('Success!');
      history.push(`/dogs/`);
    } catch {
      alert('Failed');
    }
  };

  if (loading) return <h1>Dog Is Loading</h1>;

  return (
    <div>
      <DogDetail {...dog} handleEdit={handleEdit} handleDelete={handleDelete} />
    </div>
  );
}
