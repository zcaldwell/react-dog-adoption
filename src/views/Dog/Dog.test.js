import { render, screen } from '@testing-library/react';
import Dog from './Dog';

test('dog should display dog details', async () => {
  const { container } = render(<Dog match={{ params: { id: 2 } }} />);
  await screen.findByText('Amely');
  expect(container).toMatchSnapshot();
});
