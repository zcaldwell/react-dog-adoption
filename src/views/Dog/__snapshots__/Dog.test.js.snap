// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit should display Edit details 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <h1>
        Zach
      </h1>
      <img
        class="detail-image"
        src="https://placedog.net/1000/563?id=160"
      />
      <p>
        Zach
         is 
        a 10 year
         old 
        Cocker-spaniel
      </p>
      <p>
        Yennefer hasn't aged a day and is as quick as ever. Loses things and will spend all day looking for it. 
      </p>
      <div>
        <button>
          Edit
        </button>
        <button>
          Delete
        </button>
      </div>
    </div>
  </div>
</div>
`;
